src/main/java/com/teragrep/pth_10/ast/time/DPLTimestampString.java

```
@@ -0,0 +1,190 @@
+ * modifications.
+ *
+ * Supplemented terms under GNU Affero General Public License version 3
+ * section 7
+ *
+ * Origin of the software must be attributed to Suomen Kanuuna Oy. Any modified
+ * versions must be marked as "Modified version of" The Program.
+ *
+ * Names of the licensors and authors may not be used for publicity purposes.
+ *
+ * No rights are granted for use of trade names, trademarks, or service marks
+ * which are in The Program if any.
+ *
+ * Licensee must indemnify licensors and authors for any liability that these
+ * contractual assumptions impose on licensors and authors.
+ *
+ * To the extent this program is licensed as part of the Commercial versions of
+ * Teragrep, the applicable Commercial License may apply to this file if you as
+ * a licensee so wish it.
+ */
+package com.teragrep.pth_10.ast.time;
+
+import com.teragrep.pth_10.ast.time.formats.DPLTimeFormat;
+import com.teragrep.pth_10.ast.time.formats.DefaultTimeFormat;
+import com.teragrep.pth_10.ast.time.formats.EpochSecondsTimeFormat;
+import com.teragrep.pth_10.ast.time.formats.ISO8601TimeFormat;
+import com.teragrep.pth_10.ast.time.formats.ISO8601TimeFormatWithMillis;
+import com.teragrep.pth_10.ast.time.formats.ISO8601TimeFormatWithZone;
+import com.teragrep.pth_10.ast.time.formats.ISO8601TimeFormatWithZoneAndMillis;
+import com.teragrep.pth_10.ast.time.formats.RelativeTimeFormat;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.time.ZonedDateTime;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+public final class DPLTimestampString {
+
+    private final Logger LOGGER = LoggerFactory.getLogger(DPLTimestampString.class);
+
+    private final String timestampString;
+    private final ZonedDateTime baseTime;
+    private final List<DPLTimeFormat> defaultFormats;
+    private final List<DPLTimeFormat> userDefinedFormats;
```

I think these do not need to be separate, because the parsing should fail if the user defined timeformat is not valid.

---

Issue was that user can provide a format where a value can match with default format and the user defined format. in that chase should we just throw an exception that they clashed?